tokio/src/runtime/builder.rs

```diff
@@ -351,48 +355,52 @@ impl Builder {
             let after_start = self.after_start.clone();
             let before_stop = self.before_stop.clone();
 
-            let mut builder = thread_pool::Builder::new();
-            builder.num_threads(self.num_threads);
-            builder.name(&self.thread_name);
-
-            if let Some(stack_size) = self.thread_stack_size {
-                builder.stack_size(stack_size);
-            }
-
-            builder
-                .around_worker(move |index, next| {
-                    // Configure the network driver
-                    let _net = io::set_default(&net_handles[index]);
-
-                    // Configure the clock
-                    clock::with_default(&clock, || {
-                        // Configure the timer
-                        let _timer = timer::set_default(&timer_handles[index]);
-
-                        // Call the start callback
-                        if let Some(after_start) = after_start.as_ref() {
-                            after_start();
-                        }
-
-                        // Run the worker
-                        next();
-
-                        // Call the after call back
-                        if let Some(before_stop) = before_stop.as_ref() {
-                            before_stop();
-                        }
-                    })
+            let around_worker = Arc::new(Box::new(move |index, next: &mut dyn FnMut()| {
+                // Configure the network driver
+                let _net = io::set_default(&net_handles[index]);
+
+                // Configure the clock
+                clock::with_default(&clock, || {
+                    // Configure the timer
+                    let _timer = timer::set_default(&timer_handles[index]);
+
+                    // Call the start callback
+                    if let Some(after_start) = after_start.as_ref() {
+                        after_start();
+                    }
+
+                    // Run the worker
+                    next();
+
+                    // Call the after call back
+                    if let Some(before_stop) = before_stop.as_ref() {
+                        before_stop();
+                    }
                 })
-                .build(move |index| timers[index].lock().unwrap().take().unwrap())
+            })
+                as Box<dyn Fn(usize, &mut dyn FnMut()) + Send + Sync>);
```

Comment:
nit: this formatting feels a little weird to me...what about using a separate let binding for the closure? maybe that wouldn't be any better.

Reply:
open issue w/ rustfmt / clippy :P

Reply:
I'm guessing this specific code won't stay around in follow up PRs, so i'm going to skip the nit.